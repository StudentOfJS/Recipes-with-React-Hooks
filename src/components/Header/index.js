import React, { memo } from "react";

export default memo(() => {
  return (
    <header>
      <h1>my recipes</h1>
    </header>
  );
});
