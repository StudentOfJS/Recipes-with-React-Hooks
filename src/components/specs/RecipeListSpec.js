import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from '../Header';

describe('<Header />', () => {
  test('should render correctly', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
