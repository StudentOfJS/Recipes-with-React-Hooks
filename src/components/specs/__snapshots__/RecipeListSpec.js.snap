// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> should not break when no props passed 1`] = `
<ul
  className="list-reset"
  role="Menu"
/>
`;

exports[`<RecipeList /> should render favorites correctly 1`] = `
<ul
  className="list-reset"
  role="Menu"
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer flex"
  >
    <span
      aria-checked={true}
      aria-label="check box to favorite recipe"
      className="checkbox-container"
      id={1}
      onClick={[Function]}
      onKeyPress={[Function]}
      role="checkbox"
      tabIndex={1}
    >
      <input
        aria-label="check box to favorite recipe"
        checked={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkmark"
      />
    </span>
    <button
      className="menu-button"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={2}
      type="button"
      value={1}
    >
      test recipe 1 - test category
    </button>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer flex"
  >
    <span
      aria-checked={false}
      aria-label="check box to favorite recipe"
      className="checkbox-container"
      id={2}
      onClick={[Function]}
      onKeyPress={[Function]}
      role="checkbox"
      tabIndex={3}
    >
      <input
        aria-label="check box to favorite recipe"
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkmark"
      />
    </span>
    <button
      className="menu-button"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={4}
      type="button"
      value={2}
    >
      test recipe 2 - test category
    </button>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer flex"
  >
    <span
      aria-checked={false}
      aria-label="check box to favorite recipe"
      className="checkbox-container"
      id={3}
      onClick={[Function]}
      onKeyPress={[Function]}
      role="checkbox"
      tabIndex={5}
    >
      <input
        aria-label="check box to favorite recipe"
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkmark"
      />
    </span>
    <button
      className="menu-button"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={6}
      type="button"
      value={3}
    >
      test recipe 3 - test category
    </button>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer flex"
  >
    <span
      aria-checked={false}
      aria-label="check box to favorite recipe"
      className="checkbox-container"
      id={4}
      onClick={[Function]}
      onKeyPress={[Function]}
      role="checkbox"
      tabIndex={7}
    >
      <input
        aria-label="check box to favorite recipe"
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkmark"
      />
    </span>
    <button
      className="menu-button"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={8}
      type="button"
      value={4}
    >
      test recipe 4 - test category
    </button>
  </li>
</ul>
`;

exports[`<RecipeList /> should render recipes correctly 1`] = `
<ul
  className="list-reset"
  role="Menu"
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer flex"
  >
    <span
      aria-checked={false}
      aria-label="check box to favorite recipe"
      className="checkbox-container"
      id={1}
      onClick={[Function]}
      onKeyPress={[Function]}
      role="checkbox"
      tabIndex={1}
    >
      <input
        aria-label="check box to favorite recipe"
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkmark"
      />
    </span>
    <button
      className="menu-button"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={2}
      type="button"
      value={1}
    >
      test recipe 1 - test category
    </button>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer flex"
  >
    <span
      aria-checked={false}
      aria-label="check box to favorite recipe"
      className="checkbox-container"
      id={2}
      onClick={[Function]}
      onKeyPress={[Function]}
      role="checkbox"
      tabIndex={3}
    >
      <input
        aria-label="check box to favorite recipe"
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkmark"
      />
    </span>
    <button
      className="menu-button"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={4}
      type="button"
      value={2}
    >
      test recipe 2 - test category
    </button>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer flex"
  >
    <span
      aria-checked={false}
      aria-label="check box to favorite recipe"
      className="checkbox-container"
      id={3}
      onClick={[Function]}
      onKeyPress={[Function]}
      role="checkbox"
      tabIndex={5}
    >
      <input
        aria-label="check box to favorite recipe"
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkmark"
      />
    </span>
    <button
      className="menu-button"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={6}
      type="button"
      value={3}
    >
      test recipe 3 - test category
    </button>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer flex"
  >
    <span
      aria-checked={false}
      aria-label="check box to favorite recipe"
      className="checkbox-container"
      id={4}
      onClick={[Function]}
      onKeyPress={[Function]}
      role="checkbox"
      tabIndex={7}
    >
      <input
        aria-label="check box to favorite recipe"
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkmark"
      />
    </span>
    <button
      className="menu-button"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={8}
      type="button"
      value={4}
    >
      test recipe 4 - test category
    </button>
  </li>
</ul>
`;
